crates/core/src/host/module_host.rs

```diff
@@ -72,7 +72,7 @@ impl DatabaseUpdate {
     }
 
     pub fn from_writes(stdb: &RelationalDB, tx_data: &TxData) -> Self {
-        let mut map: HashMap<TableId, (Vec<ProductValue>, Vec<ProductValue>)> = HashMap::new();
+        let mut map: IntMap<_, (Vec<_>, Vec<_>)> = IntMap::new();
```

Comment:
I'd ask that you retain the more precise type ascription here, as it makes the code easier to read in contexts where one doesn't have `rust-analyzer` providing inferred types.